compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -802,6 +803,137 @@ static FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
   return result;
 }
 
+/// Finds contiguous loop ranges from the given linalg.generic.
+static SmallVector<ReassociationIndices> getContiguousLoops(
+    linalg::GenericOp genericOp, SmallVectorImpl<unsigned> &loopDims) {
+  SmallVector<ReassociationIndices> contiguousLoops;
+  if (loopDims.size() < 2) return contiguousLoops;
+
+  auto isAllAffinemapsContiguous = [&](unsigned dim) {
```

Comment:
I don't understand the meaning of this lambda, why do we ever need to check `map.getDimPosition(dim) == dim`? That mean maps like `(d0, d1, d2, d3) -> (d1, d2, d3)` would never be considered?

Reply:
I catch this case at 
https://github.com/iree-org/iree/pull/11295/files#diff-c9f4338d6d7230f44399f248a85d6a631c264bf976c0c80a6d48eadcf442d7acR904-R909

I wrote this lambda to catch transpose like cases. 
For `affine_map<(d0, d1, d2, d3) -> (d0, d1, d3, d2)>`, it finds `d0-d1` is contigous, and `d3` and `d2` is not.